supertokens_python/recipe/thirdparty/providers/github.py

```diff
@@ -71,4 +76,33 @@ def Github(input: ProviderInput) -> Provider:  # pylint: disable=redefined-built
     if input.config.token_endpoint is None:
         input.config.token_endpoint = "https://github.com/login/oauth/access_token"
 
+    if input.config.validate_access_token is None:
+        input.config.validate_access_token = validate_access_token
+
     return NewProvider(input, GithubImpl)
+
+
+async def validate_access_token(
+    access_token: str, config: ProviderConfigForClient, _: Dict[str, Any]
+):
+    client_secret = "" if config.client_secret is None else config.client_secret
+    basic_auth_token = base64.b64encode(
+        f"{config.client_id}:{client_secret}".encode()
+    ).decode()
+
+    url = f"https://api.github.com/applications/{config.client_id}/token"
+    headers = {
+        "Authorization": f"Basic {basic_auth_token}",
+        "Content-Type": "application/json",
+    }
+    payload = json.dumps({"access_token": access_token})
+
+    resp = requests.post(url, headers=headers, data=payload)
+
+    if resp.status_code != 200:
+        raise ValueError("Invalid access token")
+
+    body = resp.json()
+
+    if "app" not in body or body["app"]["client_id"] != config.client_id:
```

Comment:
is `client_id` always guaranteed to be there if `app` is there? Cause if it's not always there, then accessing `body["app"]["client_id"]` will throw an error.

Reply:
Used `.get()` to avoid Exception.